bgpd/bgp_open.c

```diff
@@ -543,6 +543,25 @@ static as_t bgp_capability_as4(struct peer *peer, struct capability_header *hdr)
 	return as4;
 }
 
+static int bgp_capability_ext_message(struct peer *peer,
+				      struct capability_header *hdr)
+{
+	SET_FLAG(peer->cap, PEER_CAP_EXT_MESSAGE_RCV);
+
+	if (CHECK_FLAG(peer->cap, PEER_CAP_EXT_MESSAGE_ADV))
+		peer->max_packet_size = BGP_MAX_EXT_MESSAGE_PACKET_SIZE;
+
+	if (hdr->length != CAPABILITY_CODE_EXT_MESSAGE_LEN) {
+		flog_err(
+			EC_BGP_PKT_OPEN,
+			"%s: BGP Extended Message capability has incorrect data length %d",
+			peer->host, hdr->length);
+		return -1;
```

Comment:
Error handling after setting the flag and changing the buffer size? Can we do it first thing in the function?

Reply:
Sure, I’ll do that.